test/run-test.ps1

```diff
@@ -1,169 +1,62 @@
-            Write-Host "----- Publishing framework-dependant app built on $fullSdkTag to volume $framworkDepVol -----"
-            Try {
-                exec { docker run --rm `
-                    -v ${framworkDepVol}:"${containerRoot}volume" `
-                    $buildImage `
-                    dotnet publish -o ${containerRoot}volume
-                }
-
-                $fullRuntimeTag = Get-RuntimeTag $_.Dockerfile "runtime" $activeOS $manifestRepo
-                Write-Host "----- Testing on $fullRuntimeTag with $sdkTag framework-dependent app -----"
-                exec { docker run --rm `
-                    -v ${framworkDepVol}":${containerRoot}volume" `
-                    "$fullRuntimeTag" `
-                    dotnet "${containerRoot}volume${platformDirSeparator}test.dll"
-                }
-            }
-            Finally {
-                docker volume rm $framworkDepVol
-            }
-
-            if ($activeOS -eq "linux") {
-                $selfContainedImage = "self-contained-build-${buildImage}"
-                Write-Host "----- Creating publish-image for self-contained app built on $fullSdkTag -----"
-                Try {
-                    if ($sdkTag -like "2.0-sdk*")
-                    {
-                        # Workaround for https://github.com/dotnet/sdk/issues/1570
-                        $optionalRestoreParams = " /p:runtimeidentifier=debian.8-x64"
-                    }
-
-                    exec { (Get-Content ${testFilesPath}Dockerfile.linux.publish).
-                                Replace("{image}", $buildImage).
-                                Replace("{optionalRestoreParams}", $optionalRestoreParams) `
-                        | docker build $optionalDockerBuildArgs -t $selfContainedImage -
-                    }
+Try {
+    # Run Tests
+    if ([string]::IsNullOrWhiteSpace($Architecture)) {
+        $Architecture = "amd64"
+    }
 
-                    $selfContainedVol = "self-contained-publish-$appName"
-                    Write-Host "----- Publishing self-contained published app built on $fullSdkTag to volume $selfContainedVol using image $selfContainedImage -----"
-                    Try {
-                        $optionalPublishParams = "--no-restore"
-                        if ($sdkTag -like "1.*-sdk*")
-                        {
-                            $optionalPublishParams=""
-                        }
+    $TestFilter = "Architecture=$Architecture"
+    if (![string]::IsNullOrWhiteSpace($Filter)) {
+        $TestFilter += "&Version=$filter"
+    }
 
-                        exec { docker run --rm `
-                            -v ${selfContainedVol}":${containerRoot}volume" `
-                            $selfContainedImage `
-                            dotnet publish -r debian.8-x64 -o ${containerRoot}volume $optionalPublishParams
-                        }
+    & $DotnetInstallDir/dotnet test -v n --filter """$TestFilter"""
```

Comment:
`""$TestFilter"""` do we need all those quotes?

Reply:
$TestFilter needs to be quoted and quotes within quotes need to be escaped.  Escaping can be done with "" or '".